### ISSUE TITLE
Civ play- and winrates and map stats from aoestats.io

### ISSUE THREAD

**Issue by denniske** [denniske/aoe2companion#24]

**Comment by subbramanil** [comment]
Hi @denniske, 
I would like to take a stab at this issue/feature for the hacktoberfest. AOE2 has been my most favorite game since my childhood, would love to help. Can you assign it to me.

**Comment by denniske** [comment]
Hi @subbramanil, I have assigned you to this issue. If you need any help setting up the project on your computer let me know. You can reach me (denniske) in discord: [https://discord.gg/gCunWKx](https://discord.gg/gCunWKx)

**Comment by subbramanil** [comment]
I set up the environment and played around a bit. I would like to know how you would like to go about this feature. I saw that the aoe2 build order guide website was linked with the app. I feel We could do the same for the aoestats.io as well. In that per se, We don't need to re-invent the wheel, i.e., build the screen/feature from the ground up. 

Something I tried out:

<img width="709" alt="Screen Shot 2020-09-26 at 7 31 33 PM" src="https://user-images.githubusercontent.com/9157911/94352798-29d1a480-002f-11eb-8cdd-dd5b111bf276.png">

Note: The hamburger menu is the questionable one. Bit weird to see inside an app.

**Comment by denniske** [comment]
Yeah that should be easy to implement and it looks good.

We just need to inject some javascript / css to hide the 'buymeacoffee' link as donations that do not use in-app payments are a problem with iOS:
![image](https://user-images.githubusercontent.com/1076211/94353034-c49ca400-006c-11eb-8dcb-f0256e59f648.png)

**Comment by denniske** [comment]
I also think the burger menu looks okay. As there is no other burger menu in the app there won't be confusion.

**Comment by denniske** [comment]
@jerkeeler is that okay for you that we embed your site in the app?

**Comment by jerkeeler** [comment]
Hmmmmm. I would be ok with it as long as it's prominent that you are displaying aoestats.io and it's not part of your app per se. If I'm reading the screenshots correctly it looks like the url is displayed at the to of the frame, is that correct? If so I'm OK with it.

I'm kind of sad that the donation button has to be hidden, but I understand App Store policies are what they are. 

So yes, I'm OK with it as long as I get proper crediting.

**Comment by subbramanil** [comment]
@jerkeeler Appreciate for letting us embed your site in the app. Definitely credit where credit is due.

@denniske This is the preview of the screen after hiding the donation button.

<img width="423" alt="Screen Shot 2020-09-28 at 8 52 36 PM" src="https://user-images.githubusercontent.com/9157911/94638778-5b9d7200-02a0-11eb-8521-f9754cfb0d92.png">

**Comment by denniske** [comment]
@jerkeeler Yes, the url to your site will be displayed in the header. It will open your site in the browser when clicked. Thank you for letting us use your app.

@subbramanil Your preview looks good. I would accept a pull request with this design. 

For the long term there is also the possibility to query the statistics from my new match database.
Schema: [/graph/prisma/schema.prisma](/denniske/aoe2companion/blob/master/graph/prisma/schema.prisma)
Sample for getting user match stats: [/graph/src/resolver/profile.ts#L190](/denniske/aoe2companion/blob/master/graph/src/resolver/profile.ts#L190)
If you are interested in this I can provide you a database with test data. Would be a separate issue though I think.

**Comment by subbramanil** [comment]
Sure @denniske. I will submit a PR with the current design.

Sure, I can see what I can do for the long term solution. Please assign it to me when you create a separate issue.

**Comment by denniske** [comment]
@subbramanil I have merged your pull request. Thanks for the contribution.

Note: When I ran in the iOS simulator the injected javascript did not work for me. I had to add `onMessage` and use `onload` + `setTimeout` in the script to make it work: https://github.com/denniske/aoe2companion/commit/c12e3a4a5a73d37421d604d4405bfe5e771619dc

I will assign you when I open a ticket for the long term solution.
